### ISSUE TITLE
Misleading error message when using a named constant as a struct alignment

### ISSUE THREAD

**Issue by ChaiTRex** [rust-lang/rust#121425]
### Code

```Rust
const N: usize = 8;
#[repr(align(N))]
struct T;
```

### Current output

```Shell
error[E0693]: incorrect `repr(align)` attribute format: `align` takes exactly one argument in parentheses
 --> src/main.rs:2:8
  |
2 | #[repr(align(N))]
  |        ^^^^^^^^
```

### Desired output

```Shell
error[E????]: incorrect `repr(align)` value: alignment must be a literal integer
 --> src/main.rs:2:14
  |
2 | #[repr(align(N))]
  |              ^
```

### Rationale and extra context

`#[repr(align(N))]` does give `align` exactly one argument in parentheses, but the error implies that it doesn't.

### Rust Version

```Shell
rustc 1.78.0-nightly (b381d3ab2 2024-02-12)
binary: rustc
commit-hash: b381d3ab27f788f990551100c4425bb782d26d76
commit-date: 2024-02-12
host: x86_64-unknown-linux-gnu
release: 1.78.0-nightly
LLVM version: 17.0.6
```

**Comment by fmease** [comment]
For anyone who'd like to work on this, please also rework the diagnostic for `… #[repr(packed(N))] …`
